### DISCUSSION TITLE
part-ii/2006-19/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#212] [Question comments]
Part II 2006 Coding and Cryptography: 2.I.4G 

https://questions.tripos.org/part-ii/2006-19/

**Comment by walrus1729** [comment]
<details>
<summary>Solution</summary>

The first part is bookwork.

For the last part, we add $00, 01, 10$ to the code (we cannot add $11$, because $01111$ will then be both a codeword and two codewords next to each other, so the code would not be decipherable). To see that this is decipherable, we can work backwards; if we flip the message we receive and every codeword, this is clearly decipherable because then we have a prefix-free code (equivalently, our code is suffix-free, and therefore is decipherable for the same reason that a prefix-free code is decipherable).

</details>
